### PR TITLE
initial work for test centre reference data

### PR DESCRIPTION
**Pull request by andrewsetterfield**
# Description and relevant Jira numbers

## Pull Request checklist

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: src/common/framework/mysql/database.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export const query = async (
+  connection: mysql.Connection | mysql.Pool,
+  sql: string,
+  args?: any,
+): Promise<any[]> => new Promise((resolve, reject) => {
+  connection.query(sql, args, (err, rows) => {
+    if (err) {
+      reject(err);
+    }
+    // @ts-ignore
```
Comment (andrewsetterfield) [inline src/common/framework/mysql/database.ts]: looking at this

File: src/functions/getTestCentres/framework/database/query-builder.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import * as mysql from 'mysql2';
+
+/**
+ * Get all active test centres.
+ * @returns The examiners
+ */
+export const getTestCentres = async (): Promise<string> => {
```
Comment (matthew2564) [inline src/functions/getTestCentres/framework/database/query-builder.ts]: This function shouldn't be async, there is no asynchronous code inside it

File: src/functions/getTestCentres/framework/repositories/active-test-centres.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { query } from '../../../../common/framework/mysql/database';
+
+/**
+ * Call TARS replica for a list of all test centres
+ */
+export const findTestCentres = async (): Promise<any> => {
+  const connection: mysql.Connection = getConnection();
+
+  info('Searching for all test centres');
+  return query(connection, await getTestCentres());
```
Comment (matthew2564) [inline src/functions/getTestCentres/framework/repositories/active-test-centres.ts]: In other get instances, we also run connection.end such as here
```
export const getResult = async (
  appRef: number,
): Promise<TestResultRecord[]> => {
  const connection: mysql.Connection = getConnection();
  let batch;
  try {
    const [rows] = await connection.promise().query(
      buildGetResultQuery(appRef),
    );
    batch = rows;
  } finally {
    connection.end();
  }
  return batch;
};
```
Should we be doing same here?

The reason this method is also async as its awaiting connection.promise().query() which we don't seem to be doing, we are returning a promise which is fair enough but the await on getTestCentres is not doing anything.

Comment (andrewsetterfield) [reply src/functions/getTestCentres/framework/repositories/active-test-centres.ts]: sure, not there tho.  as a common funtion. dont want that repeated for each service

**Pull request merged**